packages/profile-sync-controller/src/controllers/user-storage/services.ts

```diff
@@ -93,6 +93,12 @@ export async function getUserStorage(
       nativeScryptCrypto,
     );
 
+    // Re-encrypt the entry if the salt is non-empty
+    const salt = encryption.getSalt(encryptedData);
+    if (salt.length) {
```

Comment:
Hmmm I wonder if we can instead use the `saltLen` field instead of getting the salt and then checking its length?
```
if(previouslyEncryptedData.Data.saltLen) {
  // re-encrypt with saltLen 0
}
```

Reply:
Oooh I realised the type might be unknown, and we don't have a nice way (yet) to parse and validate the encrypted data shape....